packages/shared/src/hooks/useViewSize.ts

```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react';
+import { useMedia } from './useMedia';
+import { laptop, laptopL, tablet } from '../styles/media';
+
+type ViewSize = {
+  isMobile: boolean;
+  isTablet: boolean;
+  isLaptop: boolean;
+  isLaptopL: boolean;
+};
+const useViewSize = (): ViewSize => {
+  const isMobile = !useMedia([tablet.replace('@media ', '')], [true], false);
+  const isTablet = useMedia([tablet.replace('@media ', '')], [true], false);
+  const isLaptop = useMedia([laptop.replace('@media ', '')], [true], false);
+  const isLaptopL = useMedia([laptopL.replace('@media ', '')], [true], false);
+
+  return useMemo(() => {
+    return {
+      isMobile,
+      isTablet,
+      isLaptop,
+      isLaptopL,
```

Comment:
My issue is that this is really inefficient because we register one media change handler for each of those `useMedia` hook when we could just evaluate everything in a single `useMedia`. Also on what @omBratteng commented, what about other breakpoints, having a bunch of different computed values again sounds inefficient. 

Do you think we could do it other way around by having something in lines of `useViewSize('laptop')` and that returns `true|false` and under the hood we sort out the parameters needed for `useMedia` which is the real part that is "heavy" and duplicated all over codebase, WDYT?

Reply:
The version with array would be the same as calling two hooks so I think just calling two hooks is better in this case and also not sure you could reuse `useMedia` in array manner.  Quick search around codebase shows only two files `MainFeedLayout` and `OnboardingHeader` where we `useMedia` two times in a single file if I looked correctly.

Also agree we should always try and do it through CSS because the JS check is not SSR compatible so it would bite us down the line. 👌

Reply:
I think most of the usages are boolean based which this the approach solves.

Reply:
I see so you propose a wrapping hook that can act on single sizes, but simply uses useMedia wrapped?
So 

```
useViewSize(ENUMLAPTOP)
```

```
// inside the hook
return useMedia(FORTHEENUM)
```

We can also simply change the useMedia to do the evals inside probably? (that way we only have the one hook)

Reply:
For sure, basically if we eliminate ugly API from useMedia we solve everything.. way to go 👌

Reply:
I see we have 2 edge-cases were we use it for other purposes so I'll still do wrapper, but differently.
Proposing change soon, thanks for the discussion.